src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,498 @@
+                  TRY            Length expr_Term CaseDef* finalizer_Term?
+                  REPEATED       Length elem_Term*
+                  BIND           Length boundName_NameRef patType_Type pat_Term
+                  ALTERNATIVE    Length alt_Term*
+                  UNAPPLY        Length fun_Term ImplicitArg* pat_Type pat_Term*
+                  EMPTYTREE
+                  SHARED                term_ASTRef
+  Application   = APPLY          Length fn_Term arg_Term*
+
+                  TYPEAPPLY      Length fn_Term arg_Type*
+  CaseDef       = CASEDEF        Length pat_Term rhs_Tree guard_Tree?
+  ImplicitArg   = IMPLICITARG           arg_Term
+  ASTRef        = Nat                          		// byte position in AST payload
+
+  Path          = Constant
+                  TERMREFdirect         sym_ASTRef
+                  TERMREFsymbol         sym_ASTRef qual_Type
+                  TERMREFpkg            fullyQualified_NameRef
+                  TERMREF               possiblySigned_NameRef qual_Type
+                  THIS                  clsRef_Type
+                  SKOLEMtype            refinedType_ASTRef
+                  SHARED                path_ASTRef
+
+
+  Constant      = UNITconst
+                  FALSEconst
+                  TRUEconst
+                  BYTEconst             Int
+                  SHORTconst            Int
+                  CHARconst             Nat
+                  INTconst              Int
+                  LONGconst             LongInt
+                  FLOATconst            Int
+                  DOUBLEconst           LongInt
+                  STRINGconst           NameRef
+                  NULLconst
+                  CLASSconst            Type
+                  ENUMconst             Path
+
+  Type          = Path
+                  TYPEREFdirect         sym_ASTRef
+                  TYPEREFsymbol         sym_ASTRef qual_Type
+                  TYPEREFpkg            fullyQualified_NameRef
+                  TYPEREF               possiblySigned_NameRef qual_Type
+                  SUPERtype      Length this_Type underlying_Type
+                  REFINEDtype    Length underlying_Type refinement_NameRef info_Type
+                  APPLIEDtype    Length tycon_Type arg_Type*
+                  TYPEBOUNDS     Length low_Type high_Type
+                  TYPEALIAS      Length alias_Type (COVARIANT | CONTRAVARIANT)?
+                  ANNOTATED      Length fullAnnotation_Term underlying_Type
+                  ANDtype        Length left_Type right_Type
+                  ORtype         Length left_Type right_Type
+                  BIND           Length boundName_NameRef bounds_Type selfRef_Type
+                                        // for type-variables defined in a type pattern
+                                        // bounds = type bounds, selfRef = reference to type itself.                  BYNAMEtype    Length underlying_Type
```

Comment:
Grr. It's Scala IDEs "helpful" comment mode that messes up things when editing comments. 
It's more than just formatting. That way, it was overlooked that the Length field of BYNAMEtype
should be dropped.